lib/membrane_live_web/channels/event_channel.ex

```diff
@@ -199,12 +200,44 @@ defmodule MembraneLiveWeb.EventChannel do
     {:noreply, socket}
   end
 
-  def handle_in("chat_message", %{"message" => message}, %{topic: "event:" <> id} = socket) do
+  def handle_in("chat_message", %{"content" => content}, %{topic: "event:" <> id} = socket) do
     email = socket.assigns.user_email
     {:ok, is_banned_from_chat} = check_if_banned_from_chat(email, id)
 
-    if not is_banned_from_chat,
-      do: broadcast(socket, "chat_message", %{"email" => email, "message" => message})
+    if not is_banned_from_chat do
+      %{metas: [%{name: name, is_auth: is_auth}]} = Presence.get_by_key(socket, email)
+
+      {:ok, is_presenter} = check_if_presenter(email, true, id)
+
+      new_offset =
+        if is_auth and is_presenter do
+          case :ets.lookup(:start_timestamps, id) do
+            [{_key, start_timestamp}] ->
+              System.monotonic_time(:millisecond) - start_timestamp
+
+            #  TODO handle messages when presenter is in the "start streaming" window
+            [] ->
+              raise "Recieved chat message from presenter when there is no presenters connected to RTC Engine process"
+          end
+        else
+          case :ets.lookup(:client_start_timestamps, id) do
+            [{_key, client_start_timestamp}] ->
+              System.monotonic_time(:millisecond) - client_start_timestamp
+
+            [] ->
+              0
+          end
+        end
```

Comment:
```suggestion
      key = if is_auth and is_presenter, do: :start_timestamps, else: :client_start_timestamps

      new_offset =
        cond do
          case :ets.lookup(key, id) do
            [{_key, start_timestamp}] ->
              System.monotonic_time(:millisecond) - start_timestamp

            #  TODO handle messages when presenter is in the "start streaming" window
            [] when is_auth and is_presenter ->
              raise "Recieved chat message from presenter when there is no presenters connected to RTC Engine process"

            [] ->
              0
          end
        end
```

Reply:
Is the `cond do` block intentional?

Reply:
Nope, it's a typo, it should be the only case, sorry for a mistake 😅